pkgs/development/compilers/dotnet/sdk/default.nix

```diff
@@ -0,0 +1,280 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, jq
+, curl
+, buildFHSUserEnv
+, writeScript
+, mkNugetDeps
+, mkNugetSource
+, dotnetPackages
+, writeText
+, strace
+, glibcLocales
+, pkgs
+, buildNetSdk
+}:
+
+let
+  binary-rpath = lib.makeLibraryPath (with pkgs; [
+    stdenv.cc.cc
+    zlib
+    curl
+    icu
+    libunwind
+    libuuid
+    openssl
+  ] ++ lib.optional stdenv.isLinux lttng-ust_2_12);
+
+  bootstrap-sdk = stdenv.mkDerivation {
+    pname = "dotnet-sdk";
+    version = "6.0.107";
+    src = with pkgs; runCommand "install-dotnet" ({
+      outputHashAlgo = "sha256";
+      outputHash = "sha256-IPQq9MyYrKhJaF85+EBVgT9+N9K1cnhBYPiEcxBbGDk=";
+      outputHashMode = "recursive";
+      nativeBuildInputs = [ curl cacert ];
+    }) ''
+    bash <(curl -sSL https://dot.net/v1/dotnet-install.sh) \
+      --version 6.0.107 --install-dir $out
```

Comment:
We're downloading 6.0.107 but compiling 6.0.108?

Reply:
I see, but if we'll need to download a prebuild SDK to bootstrap it regardless, I don't really see much benefit to the 2-stage approach, what can it provide compared to the current approach (to compensate compiling the SDK twice)?